ldelf/ta_elf.c

```diff
@@ -149,7 +149,30 @@ static TEE_Result e64_parse_ehdr(struct ta_elf *elf, Elf64_Ehdr *ehdr)
 
 	return TEE_SUCCESS;
 }
-#else /*ARM64*/
+#elif defined RV64
+static TEE_Result e64_parse_ehdr(struct ta_elf *elf, Elf64_Ehdr *ehdr)
+{
+	if (ehdr->e_ident[EI_VERSION] != EV_CURRENT ||
+	    ehdr->e_ident[EI_CLASS] != ELFCLASS64 ||
+	    ehdr->e_ident[EI_DATA] != ELFDATA2LSB ||
+	    ehdr->e_ident[EI_OSABI] != ELFOSABI_NONE ||
+	    ehdr->e_type != ET_DYN || ehdr->e_machine != EM_RISCV ||
+	    ehdr->e_phentsize != sizeof(Elf64_Phdr) ||
```

Comment:
Don't care about `e_flags`?

Reply:
Hi @jenswi-linaro, RISC-V e_flags contains four parts RVC, Float ABI, RVE and TSO(please refer to [https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-elf.adoc](url) ), but none of these four parts are relevant to the current processing logic.

Reply:
OK